### PR TITLE
updated `monitoring` composition for local development

### PR DESCRIPTION
**Pull request by pH14**
This PR makes several changes to update / standardize our local monitoring stack:
* Replaces the `opentelemetry` composition (previously backed by Jaeger) with the `monitoring` composition (Grafana + Prom + Tempo) to more closely match our production Grafana
* Adds a new `--monitoring` flag to `bin/environmentd` to more easily submit tracing data
* Lowers the Prometheus scrape interval to 5s. The added responsiveness when working locally is very nice
* Ingest Prom / Tempo / Tempo metrics-generator metrics into Prometheus, in addition to `envd` + `clusterd` metrics

These changes should be helpful both for local development, and for future experiments on the monitoring infrastructure itself.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/opentelemetry/README.md [hunk 1]
```diff
@@ -1,62 +1,3 @@
...
-After each query a `trace_id` will be emitted, which you can search for in
-[Jaeger] and see _exactly_ what work `environmentd` did.
-
-
-[OpenTelemetry]: https://opentelemetry.io/
-[`tracing`]: https://docs.rs/tracing/latest/tracing/
-[Jaeger]: https://www.jaegertracing.io/
-[mzcompose]: ../../doc/developer/mzcompose.md
-[basic-otlp-http example]: https://github.com/open-telemetry/opentelemetry-rust/tree/a767fd3a7f08f4d7312a1c0dbb5ac0580a108eb3/examples/basic-otlp-http
+This composition has been removed. Please see `../monitoring` for its replacement.
```
Comment (guswynn) [inline misc/opentelemetry/README.md:3]: i'd just delete the entire directory, and post in #eng-annouce!

**Review by guswynn — APPROVED** [review]
nice!

**Comment by guswynn** [comment]
nightly failures are on main so this is mergable

**Review by jpepin — APPROVED** [review]
Really nice work! Having this easy to use locally feels like a big step towards unlocking more o11y improvements.

**Pull request merged**